src/mirco_linearsolver.cpp

```diff
@@ -1,25 +1,29 @@
-#include <Epetra_SerialSpdDenseSolver.h>
-#include <Epetra_SerialSymDenseMatrix.h>
+#include <Teuchos_SerialDenseMatrix.hpp>
+#include <Teuchos_SerialDenseVector.hpp>
+#include <Teuchos_SerialSpdDenseSolver.hpp>
+#include <Teuchos_SerialSymDenseMatrix.hpp>
+#include <Teuchos_RCP.hpp>
 
 #include "mirco_linearsolver.h"
 
-void MIRCO::LinearSolver::Solve(Epetra_SerialSymDenseMatrix& matrix,
-    Epetra_SerialDenseMatrix& vector_x, Epetra_SerialDenseMatrix& vector_b)
+void MIRCO::LinearSolver::Solve(Teuchos::SerialSymDenseMatrix<int,double>& matrix,
+    Teuchos::SerialDenseVector<int,double>& vector_x, Teuchos::SerialDenseVector<int,double>& vector_b)
 {
-  Epetra_SerialSpdDenseSolver solver;
-  int err = solver.SetMatrix(matrix);
+  Teuchos::SerialSpdDenseSolver<int,double> solver;
+  int err = solver.setMatrix(Teuchos::rcpFromRef(matrix));
   if (err != 0)
   {
     std::cout << "Error setting matrix for linear solver (1)";
   }
 
-  err = solver.SetVectors(vector_x, vector_b);
+  err = solver.setVectors(Teuchos::rcpFromRef(vector_x), Teuchos::rcpFromRef(vector_b));
```

Comment:
Would it make sense to simply create the vectors as RCPs and pass them as such in the first place?

Reply:
That might be true. I'll discuss with @RShaw026 how to do that.

Reply:
I had a quick look. This requires further changes in some other files. We'll probably postpone that to a different PR.

Reply:
Ok with me.. can you please open an issue to make sure we don't forget?

Reply:
I will open an issue.